lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php

```diff
@@ -468,7 +471,7 @@ class ClassMetadataInfo implements ClassMetadata
      * READ-ONLY: The definition of the discriminator column used in JOINED and SINGLE_TABLE
      * inheritance mappings.
      *
-     * @var array
+     * @psalm-var array<string, mixed>
```

Comment:
```suggestion
     * @psalm-var array{name: string, type: string|class-string, length: int, columnDefinition: ?string}
```
inferred from `\Doctrine\ORM\Mapping\ClassMetadataInfo::setDiscriminatorColumn`